clash-lib/src/Clash/Driver/Manifest.hs

```diff
@@ -65,7 +72,58 @@ data Manifest
   , fileNames :: [(FilePath, ByteString)]
     -- ^ Names and hashes of all the generated files for the @TopEntity@. Hashes
     -- are SHA256.
-  } deriving (Show,Read)
+  } deriving (Show,Read,Eq)
+
+instance ToJSON Manifest where
+  toJSON (Manifest{..}) =
+    Aeson.object
+      [ "version" .= ("unstable" :: Text)
+      , "hash" .= manifestHash
+      , "flags" .= successFlags
+        -- TODO: add nested ports (i.e., how Clash split/filtered arguments)
+      , "components" .= componentNames
+      , "top_component" .= Aeson.object
+        [ "name" .= topComponent
+        , "ports_flat" .= Aeson.object
+          -- TODO: add width / type information
+          [ "in" .= [Aeson.object [ "name" .= p ] | p <- portInNames]
+          , "out" .= [Aeson.object [ "name" .= p ] | p <- portOutNames] ]
+        ]
+      , "files" .=
+        [ Aeson.object
+          [ "name" .= fName
+          , "sha256" .= Text.decodeUtf8 (Base16.encode fHash)
+            -- TODO: Add Edam like fields
+          ]
+        | (fName, fHash) <- fileNames]
+      ]
+
+instance FromJSON Manifest where
+  parseJSON = Aeson.withObject "Manifest" $ \v ->
+    let
+      topComponent = v .: "top_component"
+      portsFlat = topComponent >>= (.: "ports_flat")
+    in
+      Manifest
+        <$> v .: "hash"
+        <*> v .: "flags"
+        <*> (portsFlat >>= (.: "in") >>= (mapM (.: "name")))
+        <*> (portsFlat >>= (.: "out") >>= (mapM (.: "name")))
+        <*> v .: "components"
+        <*> (topComponent >>= (.: "name"))
+        <*> do
+              files <- v .: "files"
+              forM files $ \obj -> do
+                fName <- obj .: "name"
+                sha256 <- obj .: "sha256"
+                -- Note that we don't particularly care about hash decode
+                -- failures. Realistically it shouldn't happen, and if it does
+                -- there's almost no chance it would result in accidental caching.
+#if MIN_VERSION_base16_bytestring(1,0,0)
```

Comment:
Can we not set this bound in the `.cabal` file?

Reply:
This change just got out, so it's not on any Stack LTS yet </3

Reply:
:crying_cat_face: